lib/Runtime/Library/SharedArrayBuffer.cpp

```diff
@@ -452,12 +452,9 @@ namespace Js
             {
                 HeapDeleteArray(sharedContents->bufferLength, sharedContents->buffer);
             }
-
-            if (!isShutdown)
-            {
-                Recycler* recycler = GetType()->GetLibrary()->GetRecycler();
-                recycler->ReportExternalMemoryFree(sharedContents->bufferLength);
-            }
+                        
```

Comment:
didn't look into the detail but I guess somehow in Linux the finalGC is not called in linux.